flink-table/flink-table-planner/src/main/java/org/apache/flink/table/catalog/DatabaseCalciteSchema.java

```diff
@@ -77,6 +81,21 @@ public Table getTable(String tableName) {
 						!connectorTable.isBatch(),
 						FlinkStatistic.UNKNOWN()))
 					.orElseThrow(() -> new TableException("Cannot query a sink only table."));
+			} else if (table instanceof CatalogTable) {
+				Map<String, String> properties = ((CatalogTable) table).toProperties();
+				TableSource<?> tableSource = TableFactoryUtil.findAndCreateTableSource(properties);
+				if (!(tableSource instanceof StreamTableSource)) {
+					throw new TableException("Catalog tables support only StreamTableSource and BoundedTableSource");
```

Comment:
I didn't find `BoundedTableSource` in code base

Reply:
Yes, cause it does not exist yet.